app/src/main/res/layout/insert_data.xml

```diff
@@ -52,33 +52,33 @@
             <EditText
                 android:id="@+id/uid"
                 android:layout_width="match_parent"
-                android:layout_height="50dp"
-                android:layout_margin="45dp"
+                android:layout_height="@dimen/insert_height"
+                android:layout_margin="@dimen/insert_margin"
                 android:hint="@string/user_id" />
 
             <TextView
                 android:layout_width="match_parent"
                 android:layout_height="match_parent"
-                android:layout_marginStart="45dp"
-                android:layout_marginLeft="45dp"
-                android:layout_marginBottom="10dp"
+                android:layout_marginStart="@dimen/insert_margin_start"
+                android:layout_marginLeft="@dimen/insert_margin_left"
+                android:layout_marginBottom="@dimen/insert_margin_bottom"
                 android:text="@string/select_pooja_type"
-                android:textSize="15sp" />
+                android:textSize="@dimen/insert_text" />
 
             <EditText
                 android:id="@+id/money_donated"
                 android:layout_width="match_parent"
-                android:layout_height="50dp"
-                android:layout_margin="45dp"
+                android:layout_height="@dimen/insert_height"
+                android:layout_margin="@dimen/insert_margin"
                 android:hint="@string/enter_money_donated" />
 
             <Spinner
                 android:id="@+id/spinner1"
                 android:layout_width="match_parent"
-                android:layout_height="50dp"
-                android:layout_marginLeft="45dp"
```

Comment:
@hridyakrishna there are so many 45s being used as margins. Make a common thing like `margin` as the resource instead of `insert_margin_left` or `insert_margin_right` etc. Do this for other resources too.